pkg/controller/provider/validation.go

```diff
@@ -234,32 +235,33 @@ func (r *Reconciler) validateSecret(provider *api.Provider) (secret *core.Secret
 			})
 		}
 
-		url, _ := url.Parse(provider.Spec.URL)
-		if crt, err := util.GetTlsCertificate(url, secret); err == nil {
-			provider.Status.Fingerprint = util.Fingerprint(crt)
-		} else {
-			// When user specified they want to skip ceritificate verification we want to just
-			// inform the customer about failed connection and not about certificate.
-			// The reason behind this is that the GetTlsCertificate is our first attempt to connect
-			// to the provider and when connection fails, the customer will still get the cert
-			// error even when they don't care about certificates.
-			if vsphere.GetInsecureSkipVerifyFlag(secret) {
-				log.Error(err, "failed to connect to provider", "url", provider.Spec.URL)
+		err := r.VerifyTLSConnection(provider.Spec.URL, secret)
+		if err != nil {
+			switch {
+			case errors.Is(err, base.ErrProviderConnect):
 				provider.Status.SetCondition(libcnd.Condition{
 					Type:     ConnectionTestFailed,
 					Status:   True,
 					Reason:   Tested,
 					Category: Critical,
+					Message:  "Failed to connect to provider",
```

Comment:
Same here that we would just pass the error because that's the only difference between these conditions